### DISCUSSION TITLE
HowTo nav to a registered TabbedPage with Shell

### DISCUSSION THREAD

**Discussion by nk54** [dotnet/maui#8389] [Q&A]
Hello, there is tons of resource on how to create tabbed page with Shell.
But I don't find anything on how to register a page in code behind with Routing.RegisterRoute(nameof(MyPage), typeof(MyPage)) for TabbedPage.

My problem is I want to navigate to a TabbedPage which isn't set in the Shell FlyoutItems but registered in the ShellPage constructor. I don't have any problem with Tab inside a FlyoutItem. I'm ok to create another page with a different base type to make it work (instead of TabbedPage)

Navigating to a page registered in code behind works well if the page is a ContentPage but it doesn't work for TabbedPage. I don't know which base type to use.

Here is my navigation wokflow (simplified to avoid confusion)

![image](https://user-images.githubusercontent.com/13755220/176216116-c878ae41-9f81-4915-8ce8-d5447f27a84c.png)

Note : right now, my MAUI app have more than 50 pages (the initial Xamarin Forms app I migrate have more than 100 pages and work smooth and well). I don't want to put all of them inside the Shell if I can't register the route from code behind. Otherwise I would stick to the NavigationPage and avoid Shell even though I wanted to give it a try.

Any thought ? Thanks for your time

**Comment by ComptonAlvaro** [comment]
Did you find a solution for this? I would like to can navigate to a tabbed page too, but I don't see the way to do it.


Thanks so much.

**Reply by EmilAlipiev** [reply]
It seems like even today it is not supported  right?
